tests/Polyfill/PHPUnit4/PHPUnitCompatTrait.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
```

Comment:
Is there a reason we would use our own Polyfill instead of taking a dependency on something widely tested and maintained like https://github.com/Yoast/PHPUnit-Polyfills?

Reply:
I'd think it's worth having as little code to maintain as possible by using the polyfill, especially when there's a dependency that does exactly that.  I'd prefer it, yes.  @stobrien89, do you agree? 

Also as a side note, sorry this is taking so long.  It's a huge PR and carefully looking at every single file over and over again is very time consuming.  Thank you for being patient, and I'll do my best to prioritize it higher.  If you don't see me respond for a bit, feel free to @ me to give me a notification.  That said, I'll try to look at this every day until it's merged.

Reply:
@SamRemis @27pchrisl,

I definitely agree.  Especially with us looking forward to the next major version, having as little code to maintain as possible will be ideal.  Thanks again for all of the work on this @Zombaya and @27pchrisl!